pkg/serving/schema_handling.go

```diff
@@ -0,0 +1,31 @@
+package serving
+
+import (
+	"errors"
+	"fmt"
+	"github.com/knative/serving/pkg/client/clientset/versioned/scheme"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func UpdateGroupVersionKind(obj runtime.Object, gv schema.GroupVersion) error {
+	gvk, err := getGroupVersionKind(obj, gv)
+	if err != nil {
+		return err
+	}
+	obj.GetObjectKind().SetGroupVersionKind(*gvk)
+	return nil
+}
+
+func getGroupVersionKind(obj runtime.Object, gv schema.GroupVersion) (*schema.GroupVersionKind, error) {
```

Comment:
thinking if we'll need it somewhere else too, to make it public :thinking:

Reply:
It's specific to the Knative scheme (so not for other schemes like the core schema), and I don't yet see any other use case except for setting the GVK on an object.

I suggest to keep it private and only open it up if there is a concrete use case which requires this method.